Automattic/wp-calypso
# Purchases: update the purchases for backup product.

#### Changes proposed in this Pull Request

Add icon and update the display name for jetpack backup products. 

Before:

![Screen Shot 2019-11-08 at 3 46 24 PM](https://user-images.githubusercontent.com/115071/68486927-6e4f2880-0242-11ea-94a3-8a70677692f8.png)
![Screen Shot 2019-11-08 at 3 46 16 PM](https://user-images.githubusercontent.com/115071/68486932-7018ec00-0242-11ea-9de0-4ccf1c9dbadb.png)

After:
<img width="585" alt="Screen Shot 2019-11-08 at 4 02 32 PM" src="https://user-images.githubusercontent.com/115071/68486919-6abba180-0242-11ea-9481-b0052b20f9ee.png">

![Screen Shot 2019-11-11 at 4 28 02 PM](https://user-images.githubusercontent.com/115071/68599099-42c97980-04a0-11ea-91ed-4f75d313a96e.png)


#### Testing instructions

* Buy a jetpack backup product. in /plans/ page
* visit the plans /me/purchases and  specific product purchase page. 

Do you see the new fancy icon and product name.
